src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -567,6 +568,11 @@ public DatabaseRecord ReadDatabase(TransactionOperationContext context, string n
             var doc = Read(context, "db/" + name.ToLowerInvariant(), out etag);
             if (doc == null)
                 return null;
+            // If we are in passive state, we have prevent from tasks to be performed by this node.
+            if (_parent.CurrentState == RachisConsensus.State.Passive)
```

Comment:
Not a good idea, I think.
This is lying to the rest of the system.
And it won't help you with the state change for already running stuff, I think.

You need to be active about this and check in all the relevant locations.